app/models/user.rb

```diff
@@ -1,11 +1,23 @@
 class User < ApplicationRecord
   validates :email, uniqueness:true, presence:true
-  validates_presence_of :first_name, :last_name, :password
+  validates_presence_of :first_name, :last_name
   has_many :obituaries
 
-  has_secure_password
+  
+  has_secure_password validations: false
 
   def has_obituary?(obituary)
     obituaries.include?(obituary)
   end
+
+  def self.create_with_omniauth(auth)
+    where(email: auth.info.email).first_or_initialize do |user|
+      user.fbid = auth.uid
+      user.first_name = auth.info.first_name 
+      user.last_name = auth.info.last_name 
+      user.email = auth.info.email 
+      user.token = auth.credentials.token
```

Comment:
Awesome! Maybe as an extension, we can save their profile picture or do something else with Facebook.

Reply:
Stella similarly suggested doing something more with facebook functionality. Thats the only real reason I stored the facebook id.

Reply:
I think in a distant extension far down the road connecting with facebook friends on the app in order to be updated about deaths without having to be on facebook would be great, as well as importing obituaries and memorial posts/tributes. I'm really just looking for a way to get off facebook completely and not have to worry about missing this specific kind of news.